sql/tests: TestRandomSyntaxFunctions failed

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11016953?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11016953?buildTab=artifacts#/) on release-23.1 @ [81e44ab558b20b0ff55020ff05621c0ef43b4241](https://github.com/cockroachdb/cockroach/commits/81e44ab558b20b0ff55020ff05621c0ef43b4241):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 6189047 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:832: 45s of 5m0s: 111769 executions, 69628 successful
    rsg_test.go:832: 50s of 5m0s: 126128 executions, 78493 successful
    rsg_test.go:832: 55s of 5m0s: 136660 executions, 85111 successful
    rsg_test.go:832: 1m0s of 5m0s: 146932 executions, 91449 successful
    rsg_test.go:832: 1m5s of 5m0s: 158238 executions, 98490 successful
    rsg_test.go:832: 1m10s of 5m0s: 169771 executions, 105723 successful
    rsg_test.go:832: 1m15s of 5m0s: 184349 executions, 114905 successful
    rsg_test.go:832: 1m20s of 5m0s: 196226 executions, 122244 successful
    rsg_test.go:832: 1m25s of 5m0s: 208803 executions, 130093 successful
    rsg_test.go:832: 1m30s of 5m0s: 219805 executions, 136928 successful
    rsg_test.go:832: 1m35s of 5m0s: 232301 executions, 144781 successful
    rsg_test.go:832: 1m40s of 5m0s: 246840 executions, 153857 successful
    rsg_test.go:832: 1m45s of 5m0s: 258639 executions, 161173 successful
    rsg_test.go:832: 1m50s of 5m0s: 273102 executions, 170201 successful
    rsg_test.go:832: 1m55s of 5m0s: 285366 executions, 177840 successful
    rsg_test.go:832: 2m0s of 5m0s: 299115 executions, 186412 successful
    rsg_test.go:832: 2m5s of 5m0s: 310437 executions, 193443 successful
    rsg_test.go:832: 2m10s of 5m0s: 320473 executions, 199737 successful
    rsg_test.go:832: 2m15s of 5m0s: 329338 executions, 205142 successful
    rsg_test.go:832: 2m20s of 5m0s: 340128 executions, 211817 successful
    rsg_test.go:832: 2m25s of 5m0s: 351299 executions, 218891 successful
    rsg_test.go:832: 2m30s of 5m0s: 362323 executions, 225783 successful
    rsg_test.go:832: 2m35s of 5m0s: 374113 executions, 233161 successful
    rsg_test.go:832: 2m40s of 5m0s: 383512 executions, 238958 successful
    rsg_test.go:832: 2m45s of 5m0s: 393391 executions, 245099 successful
    rsg_test.go:832: 2m50s of 5m0s: 407068 executions, 253572 successful
    rsg_test.go:832: 2m55s of 5m0s: 419710 executions, 261462 successful
    rsg_test.go:832: 3m0s of 5m0s: 431197 executions, 268584 successful
    rsg_test.go:832: 3m5s of 5m0s: 443273 executions, 276076 successful
    rsg_test.go:832: 3m10s of 5m0s: 454205 executions, 282864 successful
    rsg_test.go:832: 3m15s of 5m0s: 464119 executions, 289098 successful
    rsg_test.go:832: 3m20s of 5m0s: 476688 executions, 296854 successful
    rsg_test.go:832: 3m25s of 5m0s: 491639 executions, 306141 successful
    rsg_test.go:832: 3m30s of 5m0s: 507502 executions, 316034 successful
    rsg_test.go:832: 3m35s of 5m0s: 519648 executions, 323647 successful
    rsg_test.go:832: 3m40s of 5m0s: 531334 executions, 330904 successful
    rsg_test.go:832: 3m45s of 5m0s: 546063 executions, 340041 successful
    rsg_test.go:832: 3m50s of 5m0s: 558206 executions, 347665 successful
    rsg_test.go:832: 3m55s of 5m0s: 573092 executions, 357014 successful
    rsg_test.go:832: 4m0s of 5m0s: 586121 executions, 365038 successful
    rsg_test.go:832: 4m5s of 5m0s: 600341 executions, 373917 successful
    rsg_test.go:832: 4m10s of 5m0s: 613452 executions, 382073 successful
    rsg_test.go:832: 4m15s of 5m0s: 624998 executions, 389270 successful
    rsg_test.go:832: 4m20s of 5m0s: 635983 executions, 396013 successful
    rsg_test.go:832: 4m25s of 5m0s: 650663 executions, 405039 successful
    rsg_test.go:832: 4m30s of 5m0s: 660539 executions, 411215 successful
    rsg_test.go:832: 4m35s of 5m0s: 674715 executions, 420176 successful
    rsg_test.go:832: 4m40s of 5m0s: 687160 executions, 427858 successful
    rsg_test.go:832: 4m45s of 5m0s: 701386 executions, 436757 successful
    rsg_test.go:832: 4m50s of 5m0s: 713152 executions, 444114 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30027

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11023533?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11023533?buildTab=artifacts#/) on release-23.1 @ [5fe2141d9f7c5e429e3e589a4dd722aa12840e53](https://github.com/cockroachdb/cockroach/commits/5fe2141d9f7c5e429e3e589a4dd722aa12840e53):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 5992127 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:832: 45s of 5m0s: 106821 executions, 66451 successful
    rsg_test.go:832: 50s of 5m0s: 119279 executions, 74279 successful
    rsg_test.go:832: 55s of 5m0s: 130665 executions, 81403 successful
    rsg_test.go:832: 1m0s of 5m0s: 142418 executions, 88740 successful
    rsg_test.go:832: 1m5s of 5m0s: 153323 executions, 95412 successful
    rsg_test.go:832: 1m10s of 5m0s: 163947 executions, 101976 successful
    rsg_test.go:832: 1m15s of 5m0s: 174049 executions, 108262 successful
    rsg_test.go:832: 1m20s of 5m0s: 184899 executions, 114946 successful
    rsg_test.go:832: 1m25s of 5m0s: 196458 executions, 122127 successful
    rsg_test.go:832: 1m30s of 5m0s: 207336 executions, 128933 successful
    rsg_test.go:832: 1m35s of 5m0s: 218988 executions, 136204 successful
    rsg_test.go:832: 1m40s of 5m0s: 229574 executions, 142735 successful
    rsg_test.go:832: 1m45s of 5m0s: 239997 executions, 149278 successful
    rsg_test.go:832: 1m50s of 5m0s: 250291 executions, 155576 successful
    rsg_test.go:832: 1m55s of 5m0s: 262988 executions, 163571 successful
    rsg_test.go:832: 2m0s of 5m0s: 276317 executions, 171892 successful
    rsg_test.go:832: 2m5s of 5m0s: 288524 executions, 179382 successful
    rsg_test.go:832: 2m10s of 5m0s: 301036 executions, 187194 successful
    rsg_test.go:832: 2m15s of 5m0s: 313692 executions, 195003 successful
    rsg_test.go:832: 2m20s of 5m0s: 326578 executions, 202998 successful
    rsg_test.go:832: 2m25s of 5m0s: 337961 executions, 210113 successful
    rsg_test.go:832: 2m30s of 5m0s: 351013 executions, 218267 successful
    rsg_test.go:832: 2m35s of 5m0s: 360327 executions, 224039 successful
    rsg_test.go:832: 2m40s of 5m0s: 370928 executions, 230542 successful
    rsg_test.go:832: 2m45s of 5m0s: 383222 executions, 238212 successful
    rsg_test.go:832: 2m50s of 5m0s: 393392 executions, 244628 successful
    rsg_test.go:832: 2m55s of 5m0s: 403474 executions, 250803 successful
    rsg_test.go:832: 3m0s of 5m0s: 414331 executions, 257661 successful
    rsg_test.go:832: 3m5s of 5m0s: 427016 executions, 265460 successful
    rsg_test.go:832: 3m10s of 5m0s: 437482 executions, 272076 successful
    rsg_test.go:832: 3m15s of 5m0s: 449081 executions, 279298 successful
    rsg_test.go:832: 3m20s of 5m0s: 461546 executions, 287000 successful
    rsg_test.go:832: 3m25s of 5m0s: 474994 executions, 295328 successful
    rsg_test.go:832: 3m30s of 5m0s: 484866 executions, 301499 successful
    rsg_test.go:832: 3m35s of 5m0s: 496933 executions, 309094 successful
    rsg_test.go:832: 3m40s of 5m0s: 507136 executions, 315345 successful
    rsg_test.go:832: 3m45s of 5m0s: 520456 executions, 323658 successful
    rsg_test.go:832: 3m50s of 5m0s: 534807 executions, 332670 successful
    rsg_test.go:832: 3m55s of 5m0s: 546028 executions, 339598 successful
    rsg_test.go:832: 4m0s of 5m0s: 554873 executions, 345088 successful
    rsg_test.go:832: 4m5s of 5m0s: 563817 executions, 350786 successful
    rsg_test.go:832: 4m10s of 5m0s: 576442 executions, 358570 successful
    rsg_test.go:832: 4m15s of 5m0s: 587329 executions, 365359 successful
    rsg_test.go:832: 4m20s of 5m0s: 600821 executions, 373731 successful
    rsg_test.go:832: 4m25s of 5m0s: 612796 executions, 381206 successful
    rsg_test.go:832: 4m30s of 5m0s: 625974 executions, 389444 successful
    rsg_test.go:832: 4m35s of 5m0s: 640731 executions, 398622 successful
    rsg_test.go:832: 4m40s of 5m0s: 650577 executions, 404700 successful
    rsg_test.go:832: 4m45s of 5m0s: 662635 executions, 412201 successful
    rsg_test.go:832: 4m50s of 5m0s: 676365 executions, 420640 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #107411 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-master]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11030301?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11030301?buildTab=artifacts#/) on release-23.1 @ [5fe2141d9f7c5e429e3e589a4dd722aa12840e53](https://github.com/cockroachdb/cockroach/commits/5fe2141d9f7c5e429e3e589a4dd722aa12840e53):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 5567156 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:832: 40s of 5m0s: 91158 executions, 56726 successful
    rsg_test.go:832: 45s of 5m0s: 101657 executions, 63313 successful
    rsg_test.go:832: 50s of 5m0s: 112134 executions, 69787 successful
    rsg_test.go:832: 55s of 5m0s: 121354 executions, 75599 successful
    rsg_test.go:832: 1m0s of 5m0s: 127658 executions, 79476 successful
    rsg_test.go:832: 1m5s of 5m0s: 138361 executions, 86273 successful
    rsg_test.go:832: 1m10s of 5m0s: 151354 executions, 94326 successful
    rsg_test.go:832: 1m15s of 5m0s: 164559 executions, 102544 successful
    rsg_test.go:832: 1m20s of 5m0s: 177031 executions, 110350 successful
    rsg_test.go:832: 1m25s of 5m0s: 189788 executions, 118268 successful
    rsg_test.go:832: 1m30s of 5m0s: 199705 executions, 124476 successful
    rsg_test.go:832: 1m35s of 5m0s: 211285 executions, 131663 successful
    rsg_test.go:832: 1m40s of 5m0s: 222171 executions, 138354 successful
    rsg_test.go:832: 1m45s of 5m0s: 234177 executions, 145937 successful
    rsg_test.go:832: 1m50s of 5m0s: 243875 executions, 151989 successful
    rsg_test.go:832: 1m55s of 5m0s: 257016 executions, 160158 successful
    rsg_test.go:832: 2m0s of 5m0s: 270518 executions, 168650 successful
    rsg_test.go:832: 2m5s of 5m0s: 284839 executions, 177457 successful
    rsg_test.go:832: 2m10s of 5m0s: 297230 executions, 185262 successful
    rsg_test.go:832: 2m15s of 5m0s: 307792 executions, 191774 successful
    rsg_test.go:832: 2m20s of 5m0s: 320582 executions, 199836 successful
    rsg_test.go:832: 2m25s of 5m0s: 335543 executions, 209176 successful
    rsg_test.go:832: 2m30s of 5m0s: 346645 executions, 216030 successful
    rsg_test.go:832: 2m35s of 5m0s: 356320 executions, 222046 successful
    rsg_test.go:832: 2m40s of 5m0s: 364965 executions, 227388 successful
    rsg_test.go:832: 2m45s of 5m0s: 375168 executions, 233819 successful
    rsg_test.go:832: 2m50s of 5m0s: 385590 executions, 240336 successful
    rsg_test.go:832: 2m55s of 5m0s: 397754 executions, 247750 successful
    rsg_test.go:832: 3m0s of 5m0s: 408213 executions, 254291 successful
    rsg_test.go:832: 3m5s of 5m0s: 416383 executions, 259403 successful
    rsg_test.go:832: 3m10s of 5m0s: 429194 executions, 267290 successful
    rsg_test.go:832: 3m15s of 5m0s: 440248 executions, 274235 successful
    rsg_test.go:832: 3m20s of 5m0s: 449896 executions, 280225 successful
    rsg_test.go:832: 3m25s of 5m0s: 462194 executions, 287889 successful
    rsg_test.go:832: 3m30s of 5m0s: 475281 executions, 295927 successful
    rsg_test.go:832: 3m35s of 5m0s: 485992 executions, 302609 successful
    rsg_test.go:832: 3m40s of 5m0s: 497363 executions, 309690 successful
    rsg_test.go:832: 3m45s of 5m0s: 508472 executions, 316672 successful
    rsg_test.go:832: 3m50s of 5m0s: 524489 executions, 326632 successful
    rsg_test.go:832: 3m55s of 5m0s: 537524 executions, 334724 successful
    rsg_test.go:832: 4m0s of 5m0s: 548424 executions, 341615 successful
    rsg_test.go:832: 4m5s of 5m0s: 562003 executions, 350088 successful
    rsg_test.go:832: 4m10s of 5m0s: 573739 executions, 357320 successful
    rsg_test.go:832: 4m15s of 5m0s: 585806 executions, 364845 successful
    rsg_test.go:832: 4m20s of 5m0s: 595566 executions, 370936 successful
    rsg_test.go:832: 4m25s of 5m0s: 607271 executions, 378191 successful
    rsg_test.go:832: 4m30s of 5m0s: 619512 executions, 385809 successful
    rsg_test.go:832: 4m35s of 5m0s: 628914 executions, 391682 successful
    rsg_test.go:832: 4m40s of 5m0s: 641798 executions, 399672 successful
    rsg_test.go:832: 4m45s of 5m0s: 651940 executions, 406081 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #107411 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-master]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11045346?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11045346?buildTab=artifacts#/) on release-23.1 @ [fe6e361e4148bfee1d6e9ad12d0dddb9cd364dd0](https://github.com/cockroachdb/cockroach/commits/fe6e361e4148bfee1d6e9ad12d0dddb9cd364dd0):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 6166487 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:832: 45s of 5m0s: 110685 executions, 68862 successful
    rsg_test.go:832: 50s of 5m0s: 123213 executions, 76690 successful
    rsg_test.go:832: 55s of 5m0s: 136667 executions, 85046 successful
    rsg_test.go:832: 1m0s of 5m0s: 149620 executions, 93075 successful
    rsg_test.go:832: 1m5s of 5m0s: 160780 executions, 99981 successful
    rsg_test.go:832: 1m10s of 5m0s: 173386 executions, 107788 successful
    rsg_test.go:832: 1m15s of 5m0s: 184470 executions, 114785 successful
    rsg_test.go:832: 1m20s of 5m0s: 197533 executions, 122868 successful
    rsg_test.go:832: 1m25s of 5m0s: 206934 executions, 128675 successful
    rsg_test.go:832: 1m30s of 5m0s: 219233 executions, 136365 successful
    rsg_test.go:832: 1m35s of 5m0s: 233389 executions, 145134 successful
    rsg_test.go:832: 1m40s of 5m0s: 245454 executions, 152678 successful
    rsg_test.go:832: 1m45s of 5m0s: 257847 executions, 160319 successful
    rsg_test.go:832: 1m50s of 5m0s: 270778 executions, 168445 successful
    rsg_test.go:832: 1m55s of 5m0s: 283285 executions, 176153 successful
    rsg_test.go:832: 2m0s of 5m0s: 296188 executions, 184162 successful
    rsg_test.go:832: 2m5s of 5m0s: 305113 executions, 189838 successful
    rsg_test.go:832: 2m10s of 5m0s: 319772 executions, 198882 successful
    rsg_test.go:832: 2m15s of 5m0s: 336058 executions, 209062 successful
    rsg_test.go:832: 2m20s of 5m0s: 347406 executions, 216133 successful
    rsg_test.go:832: 2m25s of 5m0s: 360720 executions, 224443 successful
    rsg_test.go:832: 2m30s of 5m0s: 370784 executions, 230600 successful
    rsg_test.go:832: 2m35s of 5m0s: 382049 executions, 237673 successful
    rsg_test.go:832: 2m40s of 5m0s: 393958 executions, 245094 successful
    rsg_test.go:832: 2m45s of 5m0s: 407538 executions, 253523 successful
    rsg_test.go:832: 2m50s of 5m0s: 421267 executions, 262021 successful
    rsg_test.go:832: 2m55s of 5m0s: 434145 executions, 269964 successful
    rsg_test.go:832: 3m0s of 5m0s: 444834 executions, 276658 successful
    rsg_test.go:832: 3m5s of 5m0s: 456736 executions, 284035 successful
    rsg_test.go:832: 3m10s of 5m0s: 466947 executions, 290329 successful
    rsg_test.go:832: 3m15s of 5m0s: 478369 executions, 297443 successful
    rsg_test.go:832: 3m20s of 5m0s: 492017 executions, 305968 successful
    rsg_test.go:832: 3m25s of 5m0s: 505694 executions, 314509 successful
    rsg_test.go:832: 3m30s of 5m0s: 514189 executions, 319892 successful
    rsg_test.go:832: 3m35s of 5m0s: 527225 executions, 327991 successful
    rsg_test.go:832: 3m40s of 5m0s: 541634 executions, 336907 successful
    rsg_test.go:832: 3m45s of 5m0s: 553097 executions, 344063 successful
    rsg_test.go:832: 3m50s of 5m0s: 563511 executions, 350588 successful
    rsg_test.go:832: 3m55s of 5m0s: 577233 executions, 359118 successful
    rsg_test.go:832: 4m0s of 5m0s: 591326 executions, 367741 successful
    rsg_test.go:832: 4m5s of 5m0s: 601918 executions, 374436 successful
    rsg_test.go:832: 4m10s of 5m0s: 613319 executions, 381527 successful
    rsg_test.go:832: 4m15s of 5m0s: 627132 executions, 390090 successful
    rsg_test.go:832: 4m20s of 5m0s: 641297 executions, 398853 successful
    rsg_test.go:832: 4m25s of 5m0s: 657517 executions, 408955 successful
    rsg_test.go:832: 4m30s of 5m0s: 671231 executions, 417439 successful
    rsg_test.go:832: 4m35s of 5m0s: 683163 executions, 424776 successful
    rsg_test.go:832: 4m40s of 5m0s: 696295 executions, 432997 successful
    rsg_test.go:832: 4m45s of 5m0s: 710718 executions, 441921 successful
    rsg_test.go:832: 4m51s of 5m0s: 723258 executions, 449650 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #107411 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-master]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11061503?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11061503?buildTab=artifacts#/) on release-23.1 @ [96579474ba06b7dd7a06d447ed19c9b052b164ec](https://github.com/cockroachdb/cockroach/commits/96579474ba06b7dd7a06d447ed19c9b052b164ec):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 5961691 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:832: 45s of 5m0s: 114092 executions, 71050 successful
    rsg_test.go:832: 50s of 5m0s: 125178 executions, 78018 successful
    rsg_test.go:832: 55s of 5m0s: 138862 executions, 86448 successful
    rsg_test.go:832: 1m0s of 5m0s: 152479 executions, 94919 successful
    rsg_test.go:832: 1m5s of 5m0s: 166334 executions, 103579 successful
    rsg_test.go:832: 1m10s of 5m0s: 179884 executions, 111964 successful
    rsg_test.go:832: 1m15s of 5m0s: 190733 executions, 118656 successful
    rsg_test.go:832: 1m20s of 5m0s: 202286 executions, 125864 successful
    rsg_test.go:832: 1m25s of 5m0s: 214791 executions, 133698 successful
    rsg_test.go:832: 1m30s of 5m0s: 228231 executions, 142116 successful
    rsg_test.go:832: 1m35s of 5m0s: 238783 executions, 148613 successful
    rsg_test.go:832: 1m40s of 5m0s: 248866 executions, 154887 successful
    rsg_test.go:832: 1m45s of 5m0s: 260057 executions, 161827 successful
    rsg_test.go:832: 1m50s of 5m0s: 269695 executions, 167868 successful
    rsg_test.go:832: 1m55s of 5m0s: 280517 executions, 174613 successful
    rsg_test.go:832: 2m0s of 5m0s: 290432 executions, 180742 successful
    rsg_test.go:832: 2m5s of 5m0s: 302333 executions, 188186 successful
    rsg_test.go:832: 2m10s of 5m0s: 312414 executions, 194459 successful
    rsg_test.go:832: 2m15s of 5m0s: 322770 executions, 200851 successful
    rsg_test.go:832: 2m20s of 5m0s: 334098 executions, 207926 successful
    rsg_test.go:832: 2m25s of 5m0s: 343596 executions, 213878 successful
    rsg_test.go:832: 2m30s of 5m0s: 354380 executions, 220567 successful
    rsg_test.go:832: 2m35s of 5m0s: 366387 executions, 228102 successful
    rsg_test.go:832: 2m40s of 5m0s: 377925 executions, 235303 successful
    rsg_test.go:832: 2m45s of 5m0s: 390080 executions, 242771 successful
    rsg_test.go:832: 2m50s of 5m0s: 403528 executions, 251127 successful
    rsg_test.go:832: 2m55s of 5m0s: 414978 executions, 258234 successful
    rsg_test.go:832: 3m0s of 5m0s: 424637 executions, 264286 successful
    rsg_test.go:832: 3m5s of 5m0s: 435431 executions, 271095 successful
    rsg_test.go:832: 3m10s of 5m0s: 447836 executions, 278770 successful
    rsg_test.go:832: 3m15s of 5m0s: 461762 executions, 287466 successful
    rsg_test.go:832: 3m20s of 5m0s: 475257 executions, 295812 successful
    rsg_test.go:832: 3m25s of 5m0s: 490097 executions, 305034 successful
    rsg_test.go:832: 3m30s of 5m0s: 499287 executions, 310710 successful
    rsg_test.go:832: 3m35s of 5m0s: 511669 executions, 318423 successful
    rsg_test.go:832: 3m40s of 5m0s: 522964 executions, 325403 successful
    rsg_test.go:832: 3m45s of 5m0s: 534487 executions, 332549 successful
    rsg_test.go:832: 3m50s of 5m0s: 545270 executions, 339254 successful
    rsg_test.go:832: 3m55s of 5m0s: 558935 executions, 347750 successful
    rsg_test.go:832: 4m0s of 5m0s: 572463 executions, 356191 successful
    rsg_test.go:832: 4m5s of 5m0s: 581995 executions, 362092 successful
    rsg_test.go:832: 4m10s of 5m0s: 593776 executions, 369352 successful
    rsg_test.go:832: 4m15s of 5m0s: 604730 executions, 376168 successful
    rsg_test.go:832: 4m21s of 5m0s: 619548 executions, 385429 successful
    rsg_test.go:832: 4m26s of 5m0s: 630604 executions, 392347 successful
    rsg_test.go:832: 4m31s of 5m0s: 643412 executions, 400250 successful
    rsg_test.go:832: 4m36s of 5m0s: 655067 executions, 407407 successful
    rsg_test.go:832: 4m41s of 5m0s: 668067 executions, 415569 successful
    rsg_test.go:832: 4m46s of 5m0s: 681252 executions, 423730 successful
    rsg_test.go:832: 4m51s of 5m0s: 696180 executions, 433065 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #107411 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-master]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11078503?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11078503?buildTab=artifacts#/) on release-23.1 @ [aaedb3dd0d97097cff7afeb2532c6265d3d62818](https://github.com/cockroachdb/cockroach/commits/aaedb3dd0d97097cff7afeb2532c6265d3d62818):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 5942316 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:832: 45s of 5m0s: 106411 executions, 66069 successful
    rsg_test.go:832: 50s of 5m0s: 118163 executions, 73430 successful
    rsg_test.go:832: 55s of 5m0s: 129052 executions, 80223 successful
    rsg_test.go:832: 1m0s of 5m0s: 138114 executions, 85895 successful
    rsg_test.go:832: 1m5s of 5m0s: 148206 executions, 92250 successful
    rsg_test.go:832: 1m10s of 5m0s: 160365 executions, 99826 successful
    rsg_test.go:832: 1m15s of 5m0s: 168918 executions, 105237 successful
    rsg_test.go:832: 1m20s of 5m0s: 178831 executions, 111405 successful
    rsg_test.go:832: 1m25s of 5m0s: 187582 executions, 116816 successful
    rsg_test.go:832: 1m30s of 5m0s: 198542 executions, 123634 successful
    rsg_test.go:832: 1m35s of 5m0s: 212451 executions, 132213 successful
    rsg_test.go:832: 1m40s of 5m0s: 224174 executions, 139525 successful
    rsg_test.go:832: 1m45s of 5m0s: 238867 executions, 148695 successful
    rsg_test.go:832: 1m50s of 5m0s: 252896 executions, 157462 successful
    rsg_test.go:832: 1m55s of 5m0s: 264145 executions, 164378 successful
    rsg_test.go:832: 2m0s of 5m0s: 276540 executions, 172177 successful
    rsg_test.go:832: 2m5s of 5m0s: 288728 executions, 179681 successful
    rsg_test.go:832: 2m10s of 5m0s: 299378 executions, 186368 successful
    rsg_test.go:832: 2m15s of 5m0s: 309040 executions, 192346 successful
    rsg_test.go:832: 2m20s of 5m0s: 318754 executions, 198416 successful
    rsg_test.go:832: 2m25s of 5m0s: 329151 executions, 204801 successful
    rsg_test.go:832: 2m30s of 5m0s: 340183 executions, 211659 successful
    rsg_test.go:832: 2m35s of 5m0s: 352975 executions, 219687 successful
    rsg_test.go:832: 2m40s of 5m0s: 364532 executions, 226931 successful
    rsg_test.go:832: 2m45s of 5m0s: 376137 executions, 234122 successful
    rsg_test.go:832: 2m50s of 5m0s: 387684 executions, 241238 successful
    rsg_test.go:832: 2m55s of 5m0s: 399050 executions, 248335 successful
    rsg_test.go:832: 3m0s of 5m0s: 407313 executions, 253374 successful
    rsg_test.go:832: 3m5s of 5m0s: 418077 executions, 260154 successful
    rsg_test.go:832: 3m10s of 5m0s: 431754 executions, 268614 successful
    rsg_test.go:832: 3m15s of 5m0s: 442188 executions, 275049 successful
    rsg_test.go:832: 3m20s of 5m0s: 452614 executions, 281578 successful
    rsg_test.go:832: 3m25s of 5m0s: 464449 executions, 288913 successful
    rsg_test.go:832: 3m30s of 5m0s: 474620 executions, 295219 successful
    rsg_test.go:832: 3m35s of 5m0s: 487105 executions, 303076 successful
    rsg_test.go:832: 3m40s of 5m0s: 494890 executions, 307862 successful
    rsg_test.go:832: 3m45s of 5m0s: 506713 executions, 315145 successful
    rsg_test.go:832: 3m50s of 5m0s: 516462 executions, 321201 successful
    rsg_test.go:832: 3m55s of 5m0s: 528113 executions, 328373 successful
    rsg_test.go:832: 4m0s of 5m0s: 539879 executions, 335775 successful
    rsg_test.go:832: 4m5s of 5m0s: 552138 executions, 343467 successful
    rsg_test.go:832: 4m11s of 5m0s: 566261 executions, 352160 successful
    rsg_test.go:832: 4m16s of 5m0s: 575347 executions, 357926 successful
    rsg_test.go:832: 4m21s of 5m0s: 586102 executions, 364567 successful
    rsg_test.go:832: 4m26s of 5m0s: 597654 executions, 371744 successful
    rsg_test.go:832: 4m31s of 5m0s: 610942 executions, 379975 successful
    rsg_test.go:832: 4m36s of 5m0s: 621424 executions, 386559 successful
    rsg_test.go:832: 4m41s of 5m0s: 632982 executions, 393773 successful
    rsg_test.go:832: 4m46s of 5m0s: 645723 executions, 401678 successful
    rsg_test.go:832: 4m51s of 5m0s: 657861 executions, 409265 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #107411 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-master]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11094286?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11094286?buildTab=artifacts#/) on release-23.1 @ [f698eca431a12cf1526ef18ac093ecba7da48b23](https://github.com/cockroachdb/cockroach/commits/f698eca431a12cf1526ef18ac093ecba7da48b23):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 5990554 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:832: 45s of 5m0s: 103708 executions, 64628 successful
    rsg_test.go:832: 50s of 5m0s: 116815 executions, 72702 successful
    rsg_test.go:832: 55s of 5m0s: 131733 executions, 82020 successful
    rsg_test.go:832: 1m0s of 5m0s: 142059 executions, 88481 successful
    rsg_test.go:832: 1m5s of 5m0s: 152092 executions, 94683 successful
    rsg_test.go:832: 1m10s of 5m0s: 165159 executions, 102823 successful
    rsg_test.go:832: 1m15s of 5m0s: 177286 executions, 110262 successful
    rsg_test.go:832: 1m20s of 5m0s: 188787 executions, 117407 successful
    rsg_test.go:832: 1m25s of 5m0s: 201866 executions, 125574 successful
    rsg_test.go:832: 1m30s of 5m0s: 212029 executions, 131831 successful
    rsg_test.go:832: 1m35s of 5m0s: 222914 executions, 138703 successful
    rsg_test.go:832: 1m40s of 5m0s: 234808 executions, 146013 successful
    rsg_test.go:832: 1m45s of 5m0s: 246316 executions, 153161 successful
    rsg_test.go:832: 1m50s of 5m0s: 259156 executions, 161254 successful
    rsg_test.go:832: 1m55s of 5m0s: 272727 executions, 169655 successful
    rsg_test.go:832: 2m0s of 5m0s: 283638 executions, 176376 successful
    rsg_test.go:832: 2m5s of 5m0s: 295815 executions, 183959 successful
    rsg_test.go:832: 2m10s of 5m0s: 307693 executions, 191280 successful
    rsg_test.go:832: 2m15s of 5m0s: 319885 executions, 198875 successful
    rsg_test.go:832: 2m20s of 5m0s: 333393 executions, 207353 successful
    rsg_test.go:832: 2m25s of 5m0s: 346401 executions, 215502 successful
    rsg_test.go:832: 2m30s of 5m0s: 359868 executions, 223864 successful
    rsg_test.go:832: 2m35s of 5m0s: 371892 executions, 231363 successful
    rsg_test.go:832: 2m40s of 5m0s: 383907 executions, 238832 successful
    rsg_test.go:832: 2m45s of 5m0s: 396947 executions, 246946 successful
    rsg_test.go:832: 2m50s of 5m0s: 408600 executions, 254269 successful
    rsg_test.go:832: 2m55s of 5m0s: 420345 executions, 261579 successful
    rsg_test.go:832: 3m0s of 5m0s: 435692 executions, 271115 successful
    rsg_test.go:832: 3m5s of 5m0s: 447933 executions, 278707 successful
    rsg_test.go:832: 3m10s of 5m0s: 457814 executions, 284856 successful
    rsg_test.go:832: 3m15s of 5m0s: 470058 executions, 292534 successful
    rsg_test.go:832: 3m20s of 5m0s: 483298 executions, 300751 successful
    rsg_test.go:832: 3m25s of 5m0s: 494215 executions, 307449 successful
    rsg_test.go:832: 3m30s of 5m0s: 505782 executions, 314686 successful
    rsg_test.go:832: 3m35s of 5m0s: 516363 executions, 321187 successful
    rsg_test.go:832: 3m40s of 5m0s: 528098 executions, 328464 successful
    rsg_test.go:832: 3m45s of 5m0s: 540473 executions, 336227 successful
    rsg_test.go:832: 3m50s of 5m0s: 552857 executions, 343859 successful
    rsg_test.go:832: 3m55s of 5m0s: 563445 executions, 350504 successful
    rsg_test.go:832: 4m0s of 5m0s: 576442 executions, 358627 successful
    rsg_test.go:832: 4m5s of 5m0s: 590262 executions, 367213 successful
    rsg_test.go:832: 4m10s of 5m0s: 600889 executions, 373846 successful
    rsg_test.go:832: 4m15s of 5m0s: 612223 executions, 380901 successful
    rsg_test.go:832: 4m20s of 5m0s: 625011 executions, 388932 successful
    rsg_test.go:832: 4m25s of 5m0s: 640499 executions, 398552 successful
    rsg_test.go:832: 4m30s of 5m0s: 653018 executions, 406279 successful
    rsg_test.go:832: 4m35s of 5m0s: 667128 executions, 415114 successful
    rsg_test.go:832: 4m40s of 5m0s: 681304 executions, 423836 successful
    rsg_test.go:832: 4m45s of 5m0s: 693455 executions, 431419 successful
    rsg_test.go:832: 4m50s of 5m0s: 704858 executions, 438492 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #107411 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-master]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11108617?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11108617?buildTab=artifacts#/) on release-23.1 @ [8b970683af4630cbb30b4c92f0be86cfadd97993](https://github.com/cockroachdb/cockroach/commits/8b970683af4630cbb30b4c92f0be86cfadd97993):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 6030020 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:832: 45s of 5m0s: 108710 executions, 67587 successful
    rsg_test.go:832: 50s of 5m0s: 121028 executions, 75282 successful
    rsg_test.go:832: 55s of 5m0s: 130999 executions, 81465 successful
    rsg_test.go:832: 1m0s of 5m0s: 141163 executions, 87720 successful
    rsg_test.go:832: 1m5s of 5m0s: 151561 executions, 94242 successful
    rsg_test.go:832: 1m10s of 5m0s: 161766 executions, 100539 successful
    rsg_test.go:832: 1m15s of 5m0s: 174531 executions, 108522 successful
    rsg_test.go:832: 1m20s of 5m0s: 183590 executions, 114111 successful
    rsg_test.go:832: 1m25s of 5m0s: 196074 executions, 121972 successful
    rsg_test.go:832: 1m30s of 5m0s: 207412 executions, 129081 successful
    rsg_test.go:832: 1m35s of 5m0s: 218786 executions, 136175 successful
    rsg_test.go:832: 1m40s of 5m0s: 230417 executions, 143427 successful
    rsg_test.go:832: 1m45s of 5m0s: 243073 executions, 151237 successful
    rsg_test.go:832: 1m50s of 5m0s: 254778 executions, 158537 successful
    rsg_test.go:832: 1m55s of 5m0s: 267930 executions, 166717 successful
    rsg_test.go:832: 2m0s of 5m0s: 281315 executions, 175019 successful
    rsg_test.go:832: 2m5s of 5m0s: 293786 executions, 182824 successful
    rsg_test.go:832: 2m10s of 5m0s: 305542 executions, 190067 successful
    rsg_test.go:832: 2m15s of 5m0s: 316699 executions, 197049 successful
    rsg_test.go:832: 2m20s of 5m0s: 328616 executions, 204421 successful
    rsg_test.go:832: 2m25s of 5m0s: 338204 executions, 210335 successful
    rsg_test.go:832: 2m30s of 5m0s: 348442 executions, 216678 successful
    rsg_test.go:832: 2m35s of 5m0s: 362972 executions, 225698 successful
    rsg_test.go:832: 2m40s of 5m0s: 375276 executions, 233354 successful
    rsg_test.go:832: 2m45s of 5m0s: 387815 executions, 241195 successful
    rsg_test.go:832: 2m50s of 5m0s: 398538 executions, 247796 successful
    rsg_test.go:832: 2m55s of 5m0s: 409687 executions, 254807 successful
    rsg_test.go:832: 3m0s of 5m0s: 423737 executions, 263659 successful
    rsg_test.go:832: 3m5s of 5m0s: 434077 executions, 270001 successful
    rsg_test.go:832: 3m10s of 5m0s: 445149 executions, 276963 successful
    rsg_test.go:832: 3m15s of 5m0s: 458559 executions, 285296 successful
    rsg_test.go:832: 3m20s of 5m0s: 471854 executions, 293559 successful
    rsg_test.go:832: 3m25s of 5m0s: 485714 executions, 302108 successful
    rsg_test.go:832: 3m30s of 5m0s: 500727 executions, 311506 successful
    rsg_test.go:832: 3m35s of 5m0s: 514015 executions, 319785 successful
    rsg_test.go:832: 3m40s of 5m0s: 525276 executions, 326758 successful
    rsg_test.go:832: 3m45s of 5m0s: 536309 executions, 333565 successful
    rsg_test.go:832: 3m50s of 5m0s: 547183 executions, 340301 successful
    rsg_test.go:832: 3m55s of 5m0s: 559597 executions, 348098 successful
    rsg_test.go:832: 4m0s of 5m0s: 571955 executions, 355721 successful
    rsg_test.go:832: 4m5s of 5m0s: 584083 executions, 363331 successful
    rsg_test.go:832: 4m10s of 5m0s: 597226 executions, 371548 successful
    rsg_test.go:832: 4m15s of 5m0s: 609720 executions, 379422 successful
    rsg_test.go:832: 4m20s of 5m0s: 619982 executions, 385690 successful
    rsg_test.go:832: 4m25s of 5m0s: 629464 executions, 391575 successful
    rsg_test.go:832: 4m30s of 5m0s: 640442 executions, 398514 successful
    rsg_test.go:832: 4m35s of 5m0s: 650778 executions, 404867 successful
    rsg_test.go:832: 4m40s of 5m0s: 660691 executions, 411044 successful
    rsg_test.go:832: 4m45s of 5m0s: 673310 executions, 418918 successful
    rsg_test.go:832: 4m50s of 5m0s: 684725 executions, 426042 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #107411 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-master]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11114781?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11114781?buildTab=artifacts#/) on release-23.1 @ [8b970683af4630cbb30b4c92f0be86cfadd97993](https://github.com/cockroachdb/cockroach/commits/8b970683af4630cbb30b4c92f0be86cfadd97993):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 5848119 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:832: 45s of 5m0s: 107086 executions, 66450 successful
    rsg_test.go:832: 50s of 5m0s: 120056 executions, 74554 successful
    rsg_test.go:832: 55s of 5m0s: 133621 executions, 82980 successful
    rsg_test.go:832: 1m0s of 5m0s: 146980 executions, 91257 successful
    rsg_test.go:832: 1m5s of 5m0s: 159060 executions, 98705 successful
    rsg_test.go:832: 1m10s of 5m0s: 169244 executions, 105045 successful
    rsg_test.go:832: 1m15s of 5m0s: 182420 executions, 113289 successful
    rsg_test.go:832: 1m20s of 5m0s: 195439 executions, 121328 successful
    rsg_test.go:832: 1m25s of 5m0s: 209626 executions, 130227 successful
    rsg_test.go:832: 1m30s of 5m0s: 219504 executions, 136334 successful
    rsg_test.go:832: 1m35s of 5m0s: 230765 executions, 143358 successful
    rsg_test.go:832: 1m40s of 5m0s: 241755 executions, 150223 successful
    rsg_test.go:832: 1m45s of 5m0s: 252955 executions, 157058 successful
    rsg_test.go:832: 1m50s of 5m0s: 262187 executions, 162812 successful
    rsg_test.go:832: 1m55s of 5m0s: 273621 executions, 169918 successful
    rsg_test.go:832: 2m0s of 5m0s: 284066 executions, 176527 successful
    rsg_test.go:832: 2m5s of 5m0s: 295190 executions, 183388 successful
    rsg_test.go:832: 2m10s of 5m0s: 308229 executions, 191486 successful
    rsg_test.go:832: 2m15s of 5m0s: 321576 executions, 199780 successful
    rsg_test.go:832: 2m20s of 5m0s: 330504 executions, 205295 successful
    rsg_test.go:832: 2m25s of 5m0s: 343567 executions, 213516 successful
    rsg_test.go:832: 2m30s of 5m0s: 354909 executions, 220597 successful
    rsg_test.go:832: 2m35s of 5m0s: 364481 executions, 226561 successful
    rsg_test.go:832: 2m40s of 5m0s: 373133 executions, 231835 successful
    rsg_test.go:832: 2m45s of 5m0s: 384874 executions, 239114 successful
    rsg_test.go:832: 2m50s of 5m0s: 395362 executions, 245733 successful
    rsg_test.go:832: 2m55s of 5m0s: 404996 executions, 251700 successful
    rsg_test.go:832: 3m0s of 5m0s: 416442 executions, 258816 successful
    rsg_test.go:832: 3m5s of 5m0s: 424939 executions, 264062 successful
    rsg_test.go:832: 3m10s of 5m0s: 436594 executions, 271254 successful
    rsg_test.go:832: 3m15s of 5m0s: 452157 executions, 281059 successful
    rsg_test.go:832: 3m20s of 5m0s: 466259 executions, 289886 successful
    rsg_test.go:832: 3m25s of 5m0s: 479091 executions, 297865 successful
    rsg_test.go:832: 3m31s of 5m0s: 492275 executions, 306081 successful
    rsg_test.go:832: 3m36s of 5m0s: 504851 executions, 313812 successful
    rsg_test.go:832: 3m41s of 5m0s: 515441 executions, 320477 successful
    rsg_test.go:832: 3m46s of 5m0s: 529403 executions, 329203 successful
    rsg_test.go:832: 3m51s of 5m0s: 543346 executions, 337750 successful
    rsg_test.go:832: 3m56s of 5m0s: 553975 executions, 344422 successful
    rsg_test.go:832: 4m1s of 5m0s: 565234 executions, 351381 successful
    rsg_test.go:832: 4m6s of 5m0s: 576388 executions, 358320 successful
    rsg_test.go:832: 4m11s of 5m0s: 586989 executions, 364963 successful
    rsg_test.go:832: 4m16s of 5m0s: 600498 executions, 373375 successful
    rsg_test.go:832: 4m21s of 5m0s: 610549 executions, 379615 successful
    rsg_test.go:832: 4m26s of 5m0s: 623762 executions, 387798 successful
    rsg_test.go:832: 4m31s of 5m0s: 637287 executions, 396237 successful
    rsg_test.go:832: 4m36s of 5m0s: 649842 executions, 404042 successful
    rsg_test.go:832: 4m41s of 5m0s: 663519 executions, 412547 successful
    rsg_test.go:832: 4m46s of 5m0s: 675634 executions, 419990 successful
    rsg_test.go:832: 4m51s of 5m0s: 688053 executions, 427775 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #107411 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-master]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11121275?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11121275?buildTab=artifacts#/) on release-23.1 @ [8b970683af4630cbb30b4c92f0be86cfadd97993](https://github.com/cockroachdb/cockroach/commits/8b970683af4630cbb30b4c92f0be86cfadd97993):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 5985777 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:832: 45s of 5m0s: 110478 executions, 68769 successful
    rsg_test.go:832: 50s of 5m0s: 120981 executions, 75379 successful
    rsg_test.go:832: 55s of 5m0s: 130924 executions, 81573 successful
    rsg_test.go:832: 1m0s of 5m0s: 144327 executions, 89956 successful
    rsg_test.go:832: 1m5s of 5m0s: 154664 executions, 96363 successful
    rsg_test.go:832: 1m10s of 5m0s: 168376 executions, 104866 successful
    rsg_test.go:832: 1m15s of 5m0s: 179616 executions, 111864 successful
    rsg_test.go:832: 1m20s of 5m0s: 189543 executions, 118069 successful
    rsg_test.go:832: 1m25s of 5m0s: 202808 executions, 126260 successful
    rsg_test.go:832: 1m30s of 5m0s: 214623 executions, 133631 successful
    rsg_test.go:832: 1m35s of 5m0s: 226320 executions, 140911 successful
    rsg_test.go:832: 1m40s of 5m0s: 240748 executions, 149764 successful
    rsg_test.go:832: 1m45s of 5m0s: 253071 executions, 157482 successful
    rsg_test.go:832: 1m50s of 5m0s: 264152 executions, 164294 successful
    rsg_test.go:832: 1m55s of 5m0s: 277993 executions, 172980 successful
    rsg_test.go:832: 2m0s of 5m0s: 287421 executions, 178848 successful
    rsg_test.go:832: 2m5s of 5m0s: 299346 executions, 186286 successful
    rsg_test.go:832: 2m10s of 5m0s: 313508 executions, 195020 successful
    rsg_test.go:832: 2m15s of 5m0s: 324912 executions, 202083 successful
    rsg_test.go:832: 2m20s of 5m0s: 334946 executions, 208354 successful
    rsg_test.go:832: 2m25s of 5m0s: 345551 executions, 214979 successful
    rsg_test.go:832: 2m30s of 5m0s: 357079 executions, 222103 successful
    rsg_test.go:832: 2m35s of 5m0s: 367621 executions, 228637 successful
    rsg_test.go:832: 2m40s of 5m0s: 380288 executions, 236473 successful
    rsg_test.go:832: 2m45s of 5m0s: 392062 executions, 243766 successful
    rsg_test.go:832: 2m50s of 5m0s: 404477 executions, 251489 successful
    rsg_test.go:832: 2m55s of 5m0s: 419124 executions, 260557 successful
    rsg_test.go:832: 3m0s of 5m0s: 428997 executions, 266684 successful
    rsg_test.go:832: 3m5s of 5m0s: 441725 executions, 274588 successful
    rsg_test.go:832: 3m10s of 5m0s: 454012 executions, 282264 successful
    rsg_test.go:832: 3m15s of 5m0s: 463765 executions, 288339 successful
    rsg_test.go:832: 3m20s of 5m0s: 476801 executions, 296491 successful
    rsg_test.go:832: 3m25s of 5m0s: 489994 executions, 304739 successful
    rsg_test.go:832: 3m30s of 5m0s: 500132 executions, 311098 successful
    rsg_test.go:832: 3m36s of 5m0s: 513487 executions, 319442 successful
    rsg_test.go:832: 3m41s of 5m0s: 525086 executions, 326665 successful
    rsg_test.go:832: 3m46s of 5m0s: 536117 executions, 333551 successful
    rsg_test.go:832: 3m51s of 5m0s: 550509 executions, 342541 successful
    rsg_test.go:832: 3m56s of 5m0s: 563079 executions, 350325 successful
    rsg_test.go:832: 4m1s of 5m0s: 575254 executions, 358000 successful
    rsg_test.go:832: 4m6s of 5m0s: 588206 executions, 366016 successful
    rsg_test.go:832: 4m11s of 5m0s: 599972 executions, 373373 successful
    rsg_test.go:832: 4m16s of 5m0s: 613122 executions, 381537 successful
    rsg_test.go:832: 4m21s of 5m0s: 626771 executions, 390101 successful
    rsg_test.go:832: 4m26s of 5m0s: 638394 executions, 397358 successful
    rsg_test.go:832: 4m31s of 5m0s: 648592 executions, 403562 successful
    rsg_test.go:832: 4m36s of 5m0s: 659219 executions, 410191 successful
    rsg_test.go:832: 4m41s of 5m0s: 671266 executions, 417701 successful
    rsg_test.go:832: 4m46s of 5m0s: 683151 executions, 424996 successful
    rsg_test.go:832: 4m51s of 5m0s: 696307 executions, 433268 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #107411 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-master]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/roachpb/data.go
+++ b/pkg/roachpb/data.go
@@ -810,10 +810,18 @@ func (v Value) computeChecksum(key []byte) uint32 {
 // In `1:3:Float/6.28`, the `1` is the column id diff as stored, `3` is the
 // computed (i.e. not stored) actual column id, `Float` is the type, and `6.28`
 // is the encoded value.
-func (v Value) PrettyPrint() string {
+func (v Value) PrettyPrint() (ret string) {
 	if len(v.RawBytes) == 0 {
 		return "/<empty>"
 	}
+	// In certain cases untagged bytes could be malformed because they are
+	// coming from user input, in which case recover with an error instead
+	// of crashing.
+	defer func() {
+		if r := recover(); r != nil {
+			ret = fmt.Sprintf("/<err: paniced parsing with %v>", r)
+		}
+	}()
 	var buf bytes.Buffer
 	t := v.GetTag()
 	buf.WriteRune('/')

--- a/pkg/sql/check.go
+++ b/pkg/sql/check.go
@@ -434,6 +434,11 @@ func (p *planner) RevalidateUniqueConstraintsInCurrentDB(ctx context.Context) er
 		return err
 	}
 	return inDB.ForEachDescriptor(func(desc catalog.Descriptor) error {
+		// If the context is cancelled, then we should bail out, since
+		// the actual revalidate operation might not check anything.
+		if err := ctx.Err(); err != nil {
+			return err
+		}
 		tableDesc, err := catalog.AsTableDescriptor(desc)
 		if err != nil {
 			return err

--- a/pkg/sql/sem/builtins/builtins.go
+++ b/pkg/sql/sem/builtins/builtins.go
@@ -5762,9 +5762,10 @@ SELECT
 				tree.ParamType{Name: "raw_value", Typ: types.Bytes},
 			},
 			ReturnType: tree.FixedReturnType(types.String),
-			Fn: func(_ context.Context, _ *eval.Context, args tree.Datums) (tree.Datum, error) {
+			Fn: func(_ context.Context, _ *eval.Context, args tree.Datums) (d tree.Datum, err error) {
 				var v roachpb.Value
 				v.RawBytes = []byte(tree.MustBeDBytes(args[0]))
+
 				return tree.NewDString(v.PrettyPrint()), nil
 			},
 			Info:       "This function is used only by CockroachDB's developers for testing purposes.",

--- a/pkg/sql/tests/rsg_test.go
+++ b/pkg/sql/tests/rsg_test.go
@@ -346,6 +346,10 @@ func TestRandomSyntaxFunctions(t *testing.T) {
 				switch lower {
 				case "pg_sleep":
 					continue
+				case "crdb_internal.gen_rand_ident":
+					// Generates random identifiers, so a large number are dangerous and
+					// can take a long time.
+					continue
 				case "st_frechetdistance", "st_buffer":
 					// Some spatial function are slow and testing them here
 					// is not worth it.

